crates/apub/src/activities/comment/create_or_update.rs

```diff
@@ -55,15 +55,24 @@ impl CreateOrUpdateComment {
       to: vec![public()],
       object: comment.into_apub(context).await?,
       cc: maa.ccs,
-      tag: maa.tags,
+      tag: if maa.tags.is_empty() {
+        None
+      } else if maa.tags.len() == 1 {
+        let mut tags = maa.tags;
+        Some(OneOrMany::from_one(
+          tags.pop().expect("Verified there is 1 element already"),
+        ))
+      } else {
+        Some(OneOrMany::from_many(maa.tags))
+      },
```

Comment:
I dont like your OneOrMany type, because its still necessary to handle both cases separately. Besides, if Smithereen can already handle a vec for this field, then we can also use a one element vec. And if it cant handle a vec, then it wont be compatible with mastodon and will probably be changed. So I will remove this.

Reply:
It's not about whether smithereen can handle a Vec you send to it, it's about if you can handle the single item smithereen sends to you. I'm interested to see how you handle this

Reply:
Ah i see now. i told the maintainer about it, and just inserted an array in the test fields for now.

https://mastodon.madrid/@felix/107286842709260237